Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -50,38 +52,38 @@ public static class CustomEntitySearch
             }
 
             WebApiSkillResponse response = WebApiSkillHelpers.ProcessRequestRecords(skillName, requestRecords,
-                (inRecord, outRecord) => {
+                 (inRecord, outRecord) => {
                     string text = inRecord.Data["text"] as string;
                     IList<string> words = ((JArray)inRecord.Data["words"]).ToObject<List<string>>();
-                    if (words == null)
+                    Dictionary<string, string[]> synonyms = (inRecord.Data.ContainsKey("synonyms")) ? ((JContainer)inRecord.Data["synonyms"]).ToObject<Dictionary<string, string[]>>() : new Dictionary<string, string[]>();
+                    IList<string> exactMatches = (inRecord.Data.ContainsKey("exactMatches")) ? ((JArray)inRecord.Data["exactMatches"]).ToObject<List<string>>() : new List<string>();
+                    int offset = (inRecord.Data.ContainsKey("fuzzyMatchOffset") && (int)inRecord.Data["fuzzyMatchOffset"] >= 0) ? (int)inRecord.Data["fuzzyMatchOffset"] : 0;
+                    if (words.Count == 1 && words[0] == "")
                     {
```

Comment:
This check looks like "if there is only one word thats an empty string, use json config based config"

Shouldn't it be if there's no words? Also validate that the other possible input fields aren't populated as well. You can throw an error and return a 400 if there's ambiguity of intent